modality-probe-cli/src/manifest_gen/invocations.rs

```diff
@@ -513,21 +526,28 @@ impl Invocations {
         names
     }
 
-    fn next_available_global_probe_id(
-        probe_id_offset: Option<u32>,
-        components: &[ComponentDir],
-    ) -> u32 {
-        let probe_id_offset = probe_id_offset.unwrap_or(0);
+    fn generate_probe_id(
+        gen: &mut IdGen,
+        probe_name: &str,
+        existing_probe_ids: &mut Vec<u32>,
+    ) -> ProbeId {
+        let mut max_tries = std::u16::MAX;
+        loop {
+            let next_id = gen.hashed_id(probe_name);
 
-        let next_id = components
-            .iter()
-            .map(|c| c.probes.next_available_probe_id())
-            .max()
-            .unwrap_or(1);
+            let id_already_taken = existing_probe_ids.iter().any(|&id| id == next_id.get());
+            let is_valid_probe_id = modality_probe::ProbeId::new(next_id.get()).is_some();
```

Comment:
I tried to convert this to an `if let ...`
```rust
            let id_already_taken = existing_probe_ids.iter().any(|&id| id == next_id.get());
            if let Some(probe_id) = modality_probe::ProbeId::new(next_id.get()) {
                if !id_already_taken {
                    existing_probe_ids.push(next_id.get());
                    return probe_id;
                }
            }
```
 but couldn't and was surprised by the reason:

```
error[E0308]: mismatched types
   --> modality-probe-cli/src/manifest_gen/invocations.rs:542:28
    |
542 |                     return probe_id;
    |                            ^^^^^^^^ expected struct `probes::ProbeId`, found struct `modality_probe::id::ProbeId`
```
Can the CLI use the core lib's definition of probe id instead?

Reply:
Yeah I think we can get things switched over now that we've got the std-feature-gated serde impls.
Mind if we make a separate ticket for it?